### ISSUE TITLE
Array destructuring with `mixed` type

### ISSUE THREAD

**Issue by paul-sonnenschein** [kalessil/phpinspectionsea#1995]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 5.5.0 |
| Language level | 8.4 |

Current behaviour
--
With the implementation of #1989, a warning has been introduced when destructuring arrays. This even triggers for the `mixed` type, which can for example happen when storing arrays in a generic cache using some sort of get-or-set or get-or-default functionality, ensuring that only arrays are returned:

```php
<?php

function someGenericCache(): mixed
{
    return [1, 2];
}

[$a, $b] = someGenericCache(); // Critical warning
```

Expected behaviour
--
- Ideally, the behavior with the `mixed` type is configurable, allowing users to specify how strict they want to be with the `mixed` type.
- Alternatively, given that the inspection also checks for various other types of suspicious array assignments, allowing to disable suspicious array destructuring checks separately from the other potential issues detected by this inspection might be reasonable.

**Comment by i-edwards** [comment]
I'm also seeing this with `json_decode`:
```php
<?php

['a' => $a, 'b' => $b] = json_decode('{"a": 1, "b": 2}', true); 
// [EA] This assignment doesn't make any sense here, as the assigned value isn't an array. 
```

It's a tricky one as this is only valid with `associative: true`. `['a' => $a, 'b' => $b] = json_decode('{"a": 1, "b": 2}', false);` is a legitimate error.
